src/quacc/atoms/skzcam.py

```diff
@@ -1,35 +1,180 @@
+) -> tuple[Atoms, Atoms, int, NDArray, NDArray]:
+    """
+    Read the file containing the periodic slab and adsorbate (geometry optimized) and return the key information needed to create an embedded cluster in ChemShell.
+
+    Parameters
+    ----------
+    adsorbate_slab_file
+        The path to the file containing the adsorbate molecule on the surface slab. It can be in any format that ASE can read.
+    adsorbate_idx
+        The indices of the atoms that make up the adsorbate molecule.
+    slab_center_idx
+        The the adsorbate_slab_file, this gives the indices of the atoms that are at the 'center' of the slab right beneath the adsorbate.
+
+    Returns
+    -------
+    Atoms
+        The Atoms object of the adsorbate molecule.
+    Atoms
+        The Atoms object of the surface slab.
+    int
+        The index of the first atom of the slab as listed in slab_center_idx.
+    NDArray
+        The position of the center of the cluster.
+    NDArray
+        The vector from the center of the slab to the center of mass of the adsorbate.
+    """
+
+    # Get the necessary information for the cluster from a provided slab file (in any format that ASE can read)
+    adsorbate_slab = read(adsorbate_slab_file)
+
+    # Find indices (within adsorbate_slab) of the slab
+    slab_idx = [i for i, _ in enumerate(adsorbate_slab) if i not in adsorbate_idx]
+
+    # Create slab from adsorbate_slab
+    slab = adsorbate_slab[slab_idx]
+
+    # Find index of the first center atom of the slab as listed in slab_center_idx
+    slab_first_atom_idx = next(
+        index for index, x in enumerate(slab_idx) if x == slab_center_idx[0]
+    )
+
+    # Get the centre of the cluster from the atom indices
+    slab_center_position = np.zeros(3)
+    for atom_idx in slab_center_idx:
+        slab_center_position += adsorbate_slab.get_positions()[atom_idx]
+
+    slab_center_position = slab_center_position / len(slab_center_idx)
+
+    adsorbate = adsorbate_slab[adsorbate_idx]
+
+    # Get the relative distance of the adsorbate from the first center atom of the slab as defined in the slab_center_idx
+    adsorbate_com = adsorbate.get_center_of_mass()
+    adsorbate_vector_from_slab = (
+        adsorbate[0].position - adsorbate_slab[slab_center_idx[0]].position
+    )
+
+    # Add the height of the adsorbate from the slab along the z-direction relative to the first center atom of the slab as defined in the slab_center_idx
+    adsorbate_com_z_disp = (
+        adsorbate_com[2] - adsorbate_slab[slab_center_idx[0]].position[2]
+    )
+    center_position = (
+        np.array([0.0, 0.0, adsorbate_com_z_disp])
+        + slab_center_position
+        - adsorbate_slab[slab_center_idx[0]].position
+    )
+
+    return (
+        adsorbate,
+        slab,
+        slab_first_atom_idx,
+        center_position,
+        adsorbate_vector_from_slab,
+    )
+
+
+@requires(has_chemshell, "ChemShell is not installed")
+def generate_chemshell_cluster(
```

Comment:
It doesn't look like this is being tested in the CI. While the codecov test "passes", this is a bug it seems --- if you click on the report, you'll see only 80% coverage or so and that the lines below are not covered still. Please double-check that this is covered.